jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/DebugHandler.java

```diff
@@ -76,17 +95,44 @@ public boolean handle(Request request, Response response, Callback callback) thr
         finally
         {
             // TODO this should be done in a completion event
-            print(name, "RESPONSE " + response.getStatus() + (ex == null ? "" : ("/" + ex)) + " " + response.getHeaders().get(HttpHeader.CONTENT_TYPE));
+            log("<< r=%s async=false %d%n%s", rname, response.getStatus(), response.getHeaders());
```

Comment:
We have to fix the TODO.    At the time this finally block is called, the response status and headers may not be set, or we may be in a race with them being set.

Also async= doesn't mean much anymore.

So we need to wrap the callback and look at the return of the handle call to log all the possible outcomes:
 + handle == false  then `log("XX r=%s")`
 + handle == true, callback complete then `log("<< r=%s async=false ...`
 + handle == true, callback not complete then `log("|| r=%s async=true ...`

In the callback, it should also check if handling has exited and if so it would need to `log("<< r=%s ...`

Reply:
Done.